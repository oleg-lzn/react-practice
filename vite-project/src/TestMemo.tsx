import React from "react";
import { useState, useMemo } from "react";

function Memo() {
  const [search, setSearch] = useState("");
  const [count, setCount] = useState(0);

  const users = [
    { id: 1, name: "Oleg" },
    { id: 2, name: "Alex" },
    { id: 3, name: "Elena" },
    { id: 4, name: "Ivan" },
  ];

  const filteredUsers = useMemo(() => {
    return users.filter((user) => {
      console.log('Фильтрую пользователей')
      return user.name.toLowerCase().includes(search.toLowerCase());
    });
  },[search]);

  return (
    <>
     
     <label> Список клиентов 
        <input type="checkbox" />
        <input
          type="text"
          className=""
          value={search}
          onChange={(e) => setSearch(e.target.value)}
        />
      </label>
      
      <ul className="users">
        {filteredUsers.map((user) => {
          return <li key={user.id}>{user.name}</li>;
        })}
      </ul>
      <button onClick={() => setCount(count + 1)}>Клик: {count}</button>
    </>
  );
}

export default Memo;
