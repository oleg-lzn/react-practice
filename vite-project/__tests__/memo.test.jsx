import { expect, test } from "vitest";
import { render } from "@testing-library/react";
import TestMemo from "../src/TestMemo";

test("snapshot with nothing in TestMemo", () => {
  const { asFragment } = render(<TestMemo />);
  expect(asFragment()).toMatchSnapshot();
});
