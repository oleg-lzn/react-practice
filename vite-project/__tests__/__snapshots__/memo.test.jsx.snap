// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot with nothing in TestMemo 1`] = `
<DocumentFragment>
  <label>
     Список клиентов
    <input
      type="checkbox"
    />
    <input
      class=""
      type="text"
      value=""
    />
  </label>
  <ul
    class="users"
  >
    <li>
      Oleg
    </li>
    <li>
      Alex
    </li>
    <li>
      Elena
    </li>
    <li>
      Ivan
    </li>
  </ul>
  <button>
    Клик: 0
  </button>
</DocumentFragment>
`;


